### PR TITLE
Add debug output to theme rendering

### PR DESCRIPTION
**Pull request by benbalter**
Trying to debug some theme issues locally, I found this debug output helpful, and thought rather than silently swallowing errors, it'd be better to at least make warning level notices for missing files or invalid directories.

### CODE REVIEW

File: lib/jekyll/renderer.rb [hunk 1]
```diff
@@ -139,6 +139,11 @@ def place_in_layouts(content, payload, info)
       output = content.dup
       layout = site.layouts[document.data["layout"]]
 
+      if layout
+        layout_source = layout.path.start_with?(site.source) ? :site : :theme
+        Jekyll.logger.debug("Layout source:", layout_source)
```
Comment (envygeeks) [inline lib/jekyll/renderer.rb]: I'd much rather this be folded off into a new line without the variable being created for the GC to handle.

File: lib/jekyll/theme.rb [hunk 2]
```diff
@@ -6,6 +6,8 @@ class Theme
 
     def initialize(name)
       @name = name.downcase.strip
+      Jekyll.logger.debug "Theme:", name
+      Jekyll.logger.debug "Theme source:", root
       configure_sass
```
Comment (ashmaroli) [inline lib/jekyll/theme.rb]: I added one more line in my local copy to check if path to layouts was as expected in Windows:
```ruby
Jekyll.logger.debug "Layouts Source:", layouts_path
```

Comment (parkr) [inline lib/jekyll/theme.rb]: 👍

**Comment by parkr** [comment]
@benbalter I love all of this! Have a sec to fix this up?

**Comment by ashmaroli** [comment]
@parkr I've resolved the conflicts (albeit made minor changes), keeping @benbalter 's original concept intact.

**Comment by DirtyF** [comment]
@ashmaroli latest rubygem seems to check that the files mentioned in the gemspec are present in the directory, I added the default file listing so fix the `test-theme` cucumber test.

**Comment by DirtyF** [comment]
`gem build` is now checking the validity of the values in the gemspec:

```
WARNING:  See http://guides.rubygems.org/specification-reference/ for help
ERROR:  While executing gem ... (Gem::InvalidSpecificationException)
    "FIXME" or "TODO" is not a summary
    "Put your gem's website or public repo URL here." is not a valid HTTP URI
```

**Comment by ashmaroli** [comment]
@DirtyF Thank you for pushing the fix but IMHO, it should be handled in a separate PR since the error is reproducible on the `master` branch as well.

**Comment by DirtyF** [comment]
@ashmaroli fix in #6784 has been merged on master

File: lib/jekyll/renderer.rb [hunk 3]
```diff
@@ -169,12 +169,16 @@ def place_in_layouts(content, payload, info)
...
-        "in #{document.relative_path} does not exist."
-      )
+      if invalid_layout?(layout)
+        Jekyll.logger.warn(
+          "Build Warning:",
+          "Layout '#{document.data["layout"]}' requested "\
+          "in #{document.relative_path} does not exist."
+        )
+      elsif !layout.nil?
+        layout_source = layout.path.start_with?(site.source) ? :site : :theme
```
Comment (parkr) [inline lib/jekyll/renderer.rb]: What do you think about adding this method to the `Layout` class, so you could ask the layout where it's from? 😄

Comment (ashmaroli) [reply lib/jekyll/renderer.rb]: I was planning to open a PR after this merged to do just what you asked.. (but very different from what's here), I was aiming to make most minimal changes to what @benbalter had originally authored..

**Comment by DirtyF** [comment]
@jekyllbot: merge +dev

**Pull request merged**